app/src/main/java/app/grapheneos/pdfviewer/PdfViewer.java

```diff
@@ -228,6 +261,15 @@ public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceReque
                 Log.d(TAG, "path " + path);
 
                 if ("/placeholder.pdf".equals(path)) {
+                    try {
+                        if (mInputStream != null) {
+                            mInputStream.close();
```

Comment:
If close() throws an exception, `openInputStream` below won't be reached.
To avoid this issue, make a separate method
```
    void maybeCloseInputStream() {
        InputStream stream = mInputStream;
        if (stream == null) {
            return;
        }
        mInputStream = null;
        try {
            stream.close();
        } catch (IOException ignored) {} 
    }
```
`mInputStream = null;`  before `close()` is needed to avoid closing the same stream multiple times.

Call this method from here and from `onDestroy`